motion_tracker/data_setup/parse_alov_bb.py

```diff
@@ -28,45 +28,86 @@ def parse_file(bbox_dir, filename):
-            filepaths_df['directory_path'] + '/' + filepaths_df['filename'] + \
-            '/' + filepaths_df['vid_num'] + '.jpg'
-
-    # Need to provide the full relative output path to point to where to move. 
-    # The `filename` is being used as part of what to save so that images 
-    # can be identified (there are lot with the same number, so we need an
-    # identifying feature). 
-    filepaths_df['output_filepath'] = output_dir + filepaths_df['filename'] + \
-            '_' + filepaths_df['vid_num'] + '.jpg'
-
-    for input_fp, output_fp in zip(filepath_df['input_filepath'], 
-            filepath_df['output_filepath']): 
+    parsed_df['input_filepath'] = (input_dir +   
+        parsed_df['directory_path'] + '/' + 
+        parsed_df['filename'] + '/' + parsed_df['vid_num'] 
+        + '.jpg')
+    # Need to provide the full relative output path to point to 
+    # where to move. The `filename` is being used as part of 
+    # what to save so that images can be identified (there are 
+    # lot with the same number, so we need an identifying feature). 
+    parsed_df['output_filepath'] = (output_dir +
+        parsed_df['filename'] + '_' + parsed_df['vid_num'] 
+        + '.jpg')
+    
+    parsed_df['jpg_filename'] = (parsed_df['output_filepath']
+            .apply(lambda filepath: filepath.split('/')[3]))
+
+    return parsed_df
+
+def cp_files(filepaths_df): 
+    """Copy over the files given by the attributes in the dataframe. 
+
+    Args: 
+    -----
+        filepaths_df: pandas DataFrame
+            Holds 3 columns, each of which holds a piece that 
+            determines the input filepaths. We'll copy the files 
+            at these filepaths to the output_dir.
+    """
+    
+    for input_fp, output_fp in zip(filepaths_df['input_filepath'], 
+            filepaths_df['output_filepath']): 
         shutil.copy(input_fp, output_fp)
 
+def add_framepaths(save_df): 
+    """Add columns into the dataframe pointing to pairs of frames. 
+
+    For the image generator, we'll need to easily grab successive
+    frames, and have the bounding boxes for these frames readily 
+    available. To do this, we'll create columns for the filenames
+    for each frame, as well as columns for their bounding boxes. 
+    They will be differentiated by prefixes of current and previous. 
+
+    Args: 
+    ----
+        save_df: pandas DataFrame
+    """
+
+    save_df = calc_frame_pairs(save_df)
+
+def calc_frame_pairs(save_df): 
+    """Add columns denoting the filepaths of current and previous frames. 
+
+    Each video has loads of frames, where every 5th (starting with 
+    the 1st) has a bounding box. For every frame except the first 
+    and the last, we can simply add 5 to it. For the first, 
+    we need it to always be the previous frame since there's nothing
+    before it. For the last, we need it to always be the current, since 
+    there is nothing after it. 
+    """
+    pass
+
```

Comment:
This function and the one right above are where I'm planning on doing the merging/joining so that the end result of the `.csv` is that each row holds information relating to the two frames that are fed into the net together.